DOCSP-46644: Stop balancer on a per-collection basis (#670)
mongodb/docs-cluster-to-cluster-sync: source/includes/fact-mongosync-balancer.rst

.. important::

   When the source or destination cluster is a sharded cluster, you must stop 

>    the source cluster's balancer on collections within the :ref:`namespace

Reviewer: I think it might help to be a bit more clear about the distinction between the namespace filter case and the non-namespace filter case, since per-collection balancing is only an option when mongosync is run with the namespace filter. 

Maybe something like "When running mongosync with namespace filtering and the source or destination cluster is a sharded cluster, you must disable the source cluster's balancer for all collections within the namespace filter."

And then adding a note for the non-namespace filtering, eg: "If running mongosync without namespace filtering, you must stop the balancer on both clusters."

Does this make sense?

.. important::

   When the source or destination cluster is a sharded cluster, you must stop 
   the source cluster's balancer on collections within the :ref:`namespace
   filter <c2c-filtered-sync>`. You can enable the source cluster's balancer on
   the cluster level and for collections outside the namespace filter. 
   You must also stop the destination cluster's balancer.

   If you enable the source cluster's balancer but do not use a
   namespace filter, or if you disable the balancer for all
   collections within the namespace filter, ``mongosync`` fails.

   During migration, do not run the :dbcommand:`moveChunk` or 
   :dbcommand:`moveRange` commands. If you have enabled the source cluster's
   balancer, but disabled it for collections within the namespace
   filter, do not run :dbcommand:`shardCollection` on collections
   within the namespace filter.

   To stop the balancer, run the :dbcommand:`balancerStop` command 
   and wait for the command to complete.
     
   After stopping the balancer, wait fifteen minutes before
   starting ``mongosync``. This gives the cluster time to

>    finish any in progress chunk migrations.

Author: int reviewer: would love feedback on whether this would be better if taken out of the `.. important::`

Reviewer 2: I am wondering if the second and third paragraphs can be moved out of the important to below it? or simply move the final two paragraphs up (putting all of the information about stopping the balancer together) and then seeing how that reads.

.. important::

   If the source or destination cluster is a sharded cluster
   and you are not running ``mongosync`` with :ref:`namespace
   filtering <c2c-filtered-sync>`,
   you must disable the source cluster's balancer
   by running the :dbcommand:`balancerStop` command and waiting 15 minutes
   for the command to complete.

   If the source or destination cluster is a sharded cluster and you
   are running ``mongosync`` with namespace filtering, you can enable the source cluster's

>    balancer overall but you must disable it for 

Reviewer: same as other comment on "overall"

   all collections within the namespace filter. 

>    See :ref:`disabling-balancer-filtered`. You can also disable

Reviewer: ```suggestion
   See :ref:`disabling-balancer-filtered`. You can also fully disable
```

.. important::

   You must always disable the balancer on a sharded destination
   cluster by using :dbcommand:`balancerStop`. 
   After stopping the balancer, wait **fifteen minutes** before
   starting ``mongosync``. This gives the cluster time to
   finish any in-progress chunk migrations.

   If the source or destination cluster is a sharded cluster

>    and you are not running ``mongosync`` with :ref:`namespace
>    filtering <c2c-filtered-sync>`,

Reviewer: ```suggestion
   and you are running ``mongosync`` without :ref:`namespace
   filtering <c2c-filtered-sync>`,
```

   you must disable the source cluster's balancer
   by running the :dbcommand:`balancerStop` command and waiting 15 minutes
   for the command to complete.

   If the source or destination cluster is a sharded cluster and you
   are running ``mongosync`` with namespace filtering, you can 
   globally enable the source cluster's
   balancer but you must disable it for 
   all collections within the namespace filter. 
   See :ref:`disabling-balancer-filtered`. You can also fully disable

>    the source cluster's balancer.

Reviewer: Perfect! thanks for adding this.

   During migration, do not run the :dbcommand:`moveChunk` or 
   :dbcommand:`moveRange` commands. If you have enabled the source cluster's
   balancer, but disabled it for collections within the namespace

>    filter, do not run :dbcommand:`shardCollection` on collections
>    within the namespace filter. If you run :dbcommand:`shardCollection` on 
>    collections within the namespace filter during the migration, ``mongosync``

Reviewer: ```suggestion
   filter, running :dbcommand:`shardCollection` on collections within the namespace filter during the migration causes ``mongosync`` to return
```

If possible to make these two sentences into one I think it might be cleaner, but up to your judgement–if you like to be explicit "do not run X command" that's ok too.
